src/OpenTelemetry.Extensions/Trace/RateLimitingSampler.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Based on the jaeger remote sampler for Java from https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java
+
+using System.Globalization;
+using OpenTelemetry.Extensions.Internal;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Rate limiting sampler that can be used to sample traces at a constant rate.
+/// </summary>
+public class RateLimitingSampler : Sampler
+{
+    private const string SAMPLERTYPE = "ratelimiting";
+    private const string SAMPLERTYPEKEY = "sampler.type";
+    private const string SAMPLERPARAMKEY = "sampler.param";
+
+    private readonly RateLimiter rateLimiter;
+    private readonly SamplingResult onSamplingResult;
+    private readonly SamplingResult offSamplingResult;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RateLimitingSampler"/> class.
+    /// </summary>
+    /// <param name="maxTracesPerSecond">The maximum number of traces that will be emitted each second.</param>
+    public RateLimitingSampler(int maxTracesPerSecond)
+    {
+        double maxBalance = maxTracesPerSecond < 1.0 ? 1.0 : maxTracesPerSecond;
```

Comment:
maxTracesPerSecond < 1.0 -- when would this happen ever?

Reply:
As the input arg is an int, it's possible someone could pass in a value < 1. It's not gonna work though, so defaulting to 1 makes sense here.

In this scenario, I think we should log a warning to indicate the passed in value has been overriden.